packages/assets-controllers/src/AccountTrackerController.ts

```diff
@@ -150,20 +162,50 @@ export class AccountTrackerController extends BaseController<
   }
 
   /**
-   * Refreshes all accounts in the current keychain.
+   * Refreshes the balances of the accounts depending on the multi-account setting.
+   * If multi-account is disabled, only updates the selected account balance.
+   * If multi-account is enabled, updates balances for all accounts.
+   *
+   * @async
    */
   refresh = async () => {
     this.syncAccounts();
     const accounts = { ...this.state.accounts };
+    const isMultiAccountBalancesEnabled = this.getMultiAccountBalancesEnabled();
+
+    if (!isMultiAccountBalancesEnabled) {
+      const selectedAddress = this.getSelectedAddress();
+      const balance = await this.getBalanceFromChain(selectedAddress);
+      accounts[selectedAddress] = { balance: BNToHex(balance) };
+      this.update({ accounts });
+      return;
+    }
+
     for (const address in accounts) {
-      await safelyExecuteWithTimeout(async () => {
-        const balance = await query(this.ethQuery, 'getBalance', [address]);
-        accounts[address] = { balance: BNToHex(balance) };
-      });
+      const balance = await this.getBalanceFromChain(address);
+      accounts[address] = { balance: BNToHex(balance) };
     }
+
     this.update({ accounts });
   };
 
+  /**
+   * Fetches the balance of a given address from the blockchain.
+   *
+   * @async
```

Comment:
Nit: I would recommend omitting this tag. We know it's async already from the use of the `async` keyword.

For reference, the JSDoc tags we typically use in TypeScript are those that are a part of TSDoc: https://tsdoc.org/
These will get included in the `TypeDoc`-generated documentation, and picked up by most TypeScript-focused editors/IDEs.

Edit: Likewise for the use of this tag in the `refresh` JSDoc comment